src/track/trackrecord.cpp

```diff
@@ -107,24 +107,53 @@ bool copyIfNotEmpty(
 
 } // anonymous namespace
 
+bool TrackRecord::updateSourceSynchronizedAt(
+        const QDateTime& sourceSynchronizedAt) {
+    VERIFY_OR_DEBUG_ASSERT(sourceSynchronizedAt.isValid()) {
+        // Cannot be reset after it has been set at least once.
+        // This is required to prevent unintended and repeated
+        // reimporting of metadata from file tags.
+        return false;
+    }
+    if (getSourceSynchronizedAt() == sourceSynchronizedAt) {
+        return false; // unchanged
+    }
+    setSourceSynchronizedAt(sourceSynchronizedAt);
+    m_headerParsed = sourceSynchronizedAt.isValid();
+    DEBUG_ASSERT(isSourceSynchronized());
+    return true;
+}
+
+bool TrackRecord::isSourceSynchronized() const {
+    // This method cannot be used to update m_headerParsed
+    // after modifying m_sourceSynchronizedAt during a short
+    // moment of inconsistency. Otherwise the debug assertion
```

Comment:
Is "Otherwise" correct here? Or should it be "This will violate the debug assertion"

Reply:
Otherwise = if used (vs. "cannot be used")

Reply:
Now I am completely confused ...

Reply:
This method requires that both members are in sync and must only be invoked while they are consistent. This is not guaranteed while updating them. In this case we must repeat some of the code although it seems like we could simply use `m_headerParsed = isSourceSynchronized()` for this purpose.

If the comment is confusing I could delete it. But I guarantee that someone will try to do this at some point to remove redundancy.

Reply:
Unfortunately, we need to keep and deal with this redundancy for backward compatibility.

Reply:
Ah OK, I understand it now.

> If the comment is confusing I could delete it.

In general more comments is better then less. How about adding the explanations form your comments above instead of the original comment? The forbidden code snipped is the best of it.